### PR TITLE
test: Catch Out of Memory Errors during integration tests

### LINKED ISSUE #5621: Integration Tests: Fail if OOM detected
In the past we seem to often run into Out Of Memory errors for various reasons.

It would be nice to add a step to the integration tests (preferably at the end) that does the following:

```console
kubectl get event -owide | grep OOMs
```

# Expected behaviour

If OOMs happen, the output would be as follows:
```
51m         Warning   SystemOOM   node/gke-keptn-060-internal-d-default-pool-179d6422-j3wd               kubelet, gke-keptn-060-internal-d-default-pool-179d6422-j3wd   System OOM encountered, victim process: helm-service, pid: 3135306   51m          1       gke-keptn-060-internal-d-default-pool-179d6422-j3wd.16acff1b2cbcec16
22m         Warning   SystemOOM   node/gke-keptn-060-internal-d-default-pool-179d6422-j3wd               kubelet, gke-keptn-060-internal-d-default-pool-179d6422-j3wd   System OOM encountered, victim process: helm-service, pid: 3585173   22m          1       gke-keptn-060-internal-d-default-pool-179d6422-j3wd.16ad00ac92cfa5ed
4m57s       Warning   SystemOOM   node/gke-keptn-060-internal-d-default-pool-179d6422-j3wd               kubelet, gke-keptn-060-internal-d-default-pool-179d6422-j3wd   System OOM encountered, victim process: helm-service, pid: 3633324   4m57s        1       gke-keptn-060-internal-d-default-pool-179d6422-j3wd.16ad01a4c58af1db
```

Here the following information should be extracted:
Time, Process (e.g., helm-service)

## Technical details

Our go tests already provide a way to interface with K8s

# Definition of Done

- [ ] Integration tests need to fail if any OOMs happen
- [ ] Integration tests should report that an OOM happened, when, and where

### PR DESCRIPTION
**Pull request by mowies**
### This PR
- adds checking for Out of Memory event in kubernetes after every integration test (`OOMKilling`, `SystemOOM`)
- fixes a typo

Fixes #5621

### CODE REVIEW

File: test/go-tests/k8s_utils.go [hunk 1]
```diff
@@ -159,3 +160,37 @@ func WaitForDeploymentInNamespace(deploymentName, namespace string) error {
 func WaitForPodOfDeployment(deploymentName string) error {
 	return keptnkubeutils.WaitForDeploymentToBeRolledOut(false, deploymentName, GetKeptnNameSpaceFromEnv())
 }
+
+type K8SEvent struct {
+	Reason string `json:"reason"`
+	Type   string `json:"type"`
```
Comment (thisthat) [inline test/go-tests/k8s_utils.go:166]: would it make sense also to include the message?

File: test/go-tests/k8s_utils.go [hunk 2]
```diff
@@ -159,3 +160,37 @@ func WaitForDeploymentInNamespace(deploymentName, namespace string) error {
...
+	}
+
+	eventArray := K8SEventArray{}
+	err = json.Unmarshal([]byte(events), &eventArray)
+
+	oomEvents := K8SEventArray{
+		Items: []K8SEvent{},
+	}
+	for _, event := range eventArray.Items {
+		if event.Reason == "OOMKilling" || event.Reason == "SystemOOM" {
```
Comment (thisthat) [inline test/go-tests/k8s_utils.go:187]:
```suggestion
		if event.Reason == "OOMKilled" || event.Reason == "SystemOOM" {
```

Comment (mowies) [reply test/go-tests/k8s_utils.go]: changed it to `strings.Contains("OOM")`

File: test/go-tests/k8s_utils.go (hunk 1 above)
Comment (mowies) [reply test/go-tests/k8s_utils.go]: done

**Review by thisthat — APPROVED** [review]

**Review by oleg-nenashev — APPROVED** [review]
❤️

**Pull request merged**